### PR TITLE
feat: response time worst in input

### PR DESCRIPTION
**Pull request by rokamu623**
## Pre-review checklist for the PR author

- [x] I've confirmed the [contribution guidelines](https://github.com/tier4/caret/blob/main/.github/CONTRIBUTING.md).
- [x] The PR is ready for review.

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [x] The PR has been properly tested.
- [x] The PR has been reviewed.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] (Optional) The PR has been properly tested with CARET_report verification.
- [ ] There are no open discussions or they are tracked via tickets.
- [ ] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

### CODE REVIEW

File: src/caret_analyze/record/records_service/response_time.py [hunk 1]
```diff
@@ -273,6 +274,29 @@ def to_all_records(self) -> RecordsInterface:
...
+        start_timestamps: list[int] = []
+        for start_ts, end_ts in zip(self._start_timestamps, self._end_timestamps):
+            if end_ts not in end_timestamps:
+                start_timestamps.append(start_ts)
+                end_timestamps.append(end_ts)
+            else:
+                idx = end_timestamps.index(end_ts)
+                if start_ts < start_timestamps[idx]:
+                    start_timestamps[idx] = start_ts
+                    end_timestamps[idx] = end_ts
```
[the remark this answers is not in the document]
Comment (rokamu623) [reply src/caret_analyze/record/records_service/response_time.py:288]: By this process, "start_timestamps-end_timestamps pairs" with the same "end_timestamps" value are selected only if they have a smaller "start_timestamps" value.

In other words, this process selects the "worst-case" pair.

File: src/caret_analyze/record/records_service/response_time.py [hunk 2]
```diff
@@ -273,6 +274,29 @@ def to_all_records(self) -> RecordsInterface:
...
+
+        records = self._create_empty_records()
+        for start_ts, end_ts in zip(start_timestamps, end_timestamps):
+            record = {
+                self._start_column: start_ts,
+                'response_time': end_ts - start_ts
+            }
+            records.append(record)
```
[the remark this answers is not in the document]
Comment (rokamu623) [reply src/caret_analyze/record/records_service/response_time.py:296]: We should avoid using function from other class.

**Review by keita1523 — APPROVED** [review]
LGTM.

**Review by nabetetsu — APPROVED** [review]
I have confirmed that worst_in_input gots worst value among the input candidates that make up the path as expected.

LGTM :+1:

rt2-824 drawio

**Pull request merged**